include/ast/instruction.h

```diff
@@ -37,7 +37,7 @@ class Instruction {
   /// Constructor assigns the OpCode and the Offset.
   Instruction(OpCode Byte, uint32_t Off = 0) noexcept
       : Offset(Off), Code(Byte) {
-#if defined(__x86_64__) || defined(__aarch64__)
+#if defined(__x86_64__) || defined(__aarch64__) || defined(__riscv)
```

Comment:
Before adding the `__riscv` here, you should make sure that the `riscv` target supports `uint128_t`.
Otherwise, it should be fine to fall back to the `else` case.